### DISCUSSION TITLE
Self-Hosted GitLab Support

### DISCUSSION THREAD

**Discussion by mirshko** [vercel/community#337] [Idea]
Creating a discussion here to see if other people could benefit from the addition of self-hosted GitLab accounts for Vercel deployments. 

Relying the Vercel CLI and either project linking or setting the team and project ids manually per deploy is quite a nuisance, same with setting up new GitLab pipelines and setting up runners to handle running the workflow for each repository. 

Having Vercel support self-hosted GitLab instances would allow for the same benefits that GitHub and Non-Self-Hosted GitLab users enjoy.

Would love to hear if anyone else would like this feature added!

**Comment by swarnava** [comment]
Hi there,

At the moment self-hosted gitlab is not supported on Vercel but it is a requested feature that we are working on implementing for the future.

**Comment by mozzius** [comment]
I'd love to see this! It's a bit of a hassle at the moment using pipelines and such, would be great to just connect them.

**Comment by leerob** [comment]
**Update:** It's now much easier to deploy to Vercel with any custom CI/CD provider like self-managed GitLab with GitLab Pipelines!

[We've created a guide](https://vercel.com/support/articles/how-can-i-use-gitlab-pipelines-with-vercel) and [an example](https://github.com/vercel/examples/tree/main/ci-cd/gitlab-cicd) using GitLab Pipelines. Hope this helps! 🙏

**Reply by L-U-C-K-Y** [reply]
Thanks for the update. Having used native gitlab.com integration and self-hosted, I miss a lot of features such as auto-comment on MR and auto build.
